clients/admin-ui/src/features/datastore-connections/ConnectionGridItem.tsx

```diff
@@ -45,6 +47,15 @@ const ConnectionGridItem: React.FC<ConnectionGridItemProps> = ({
   connectionData,
 }) => {
   const [trigger, result] = useLazyGetDatastoreConnectionStatusQuery();
+  const { connectionOptions } = useAppSelector(selectConnectionTypeState);
+
+  const connectionType =
```

Comment:
this is a good candidate for memoization, otherwise we'll be doing this `find` on every render. something like:

```typescript
const connectionType = useMemo(() => {
  return connectionOptions.find(...)
}, [connectionData, connectionOptions])


```